tick/preprocessing/longitudinal_features_lagger.py

```diff
@@ -175,36 +169,57 @@ def transform(self, features, labels=None, censoring=None):
         base_shape = (self._n_intervals, self._n_init_features)
         features = check_longitudinal_features_consistency(
             features, base_shape, "float64")
-        if sps.issparse(features[0]):
-            X_with_lags = [
-                self._sparse_lagger(x, int(censoring[i]))
-                for i, x in enumerate(features)
-            ]
-            # TODO: Don't get why int() is required here as censoring_i is uint64
-        else:
-            X_with_lags = [
-                self._dense_lagger(x, int(censoring[i]))
-                for i, x in enumerate(features)
-            ]
+
+        initializer = partial(self._inject_cpp_object,
+                              n_intervals=self._n_intervals,
+                              n_lags=self.n_lags)
+        callback = self._sparse_lagger if sps.issparse(features[0]) \
+            else self._dense_lagger
+        callback = partial(callback, n_intervals=self._n_intervals,
+                           n_output_features=self._n_output_features,
+                           n_lags=self.n_lags)
+
+        with Pool(self.n_jobs, initializer=initializer) as pool:
+            X_with_lags = pool.starmap(callback, zip(features, censoring))
 
         return X_with_lags, labels, censoring
 
-    def _dense_lagger(self, feature_matrix, censoring_i):
-        output = np.zeros((self._n_intervals, self._n_output_features),
-                          dtype="float64")
-        self._cpp_preprocessor.dense_lag_preprocessor(feature_matrix, output,
-                                                      censoring_i)
+    @staticmethod
+    def _inject_cpp_object(n_intervals, n_lags):
+        """Creates a global instance of the CPP preprocessor object.
+
+        WARNING: to be used only as a multiprocessing.Pool initializer.
+        In multiprocessing context, each process has its own namespace, so using
+        global is not as bad as it seems. Still, it requires to proceed with
+        caution.
+        """
+        global _cpp_preprocessor
+        _cpp_preprocessor = _LongitudinalFeaturesLagger(n_intervals, n_lags)
```

Comment:
Here is a clever way (using `__call__` in a class) to avoid global variable, did you give it a try?

https://stackoverflow.com/a/37232918/2015762